Mathlib/Topology/ProperMap.lean

```diff
@@ -226,6 +281,36 @@ theorem isProperMap_iff_isClosedMap_and_compact_fibers :
     rw [← clusterPt_lift'_closure_iff]
     exact hx.mono inf_le_left
 
+/-- An injective and continuous function is proper if and only if it is close. -/
+lemma isProperMap_iff_isClosedMap_of_inj (f_cont : Continuous f) (f_inj : f.Injective) :
+    IsProperMap f ↔ IsClosedMap f := by
+  refine ⟨fun h ↦ h.isClosedMap, fun h ↦ ?_⟩
+  rw [isProperMap_iff_isClosedMap_and_compact_fibers]
+  exact ⟨f_cont, h, fun y ↦ (subsingleton_singleton.preimage f_inj).isCompact⟩
```

Comment:
Could you also deduce [Homeomorph.isProperMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ProperMap.html#Homeomorph.isProperMap) from this? I think you can safely move it down the file.